great_expectations/experimental/datasource/fabric.py

```diff
@@ -0,0 +1,394 @@
+        use_xmla: bool = False,
+    ) -> PowerBIMeasure:
+        """Adds a PowerBIMeasure asset to this datasource.
+
+        Args:
+            name: The name of this asset.
+            order_by: A list of Sorters or Sorter strings.
+            batch_metadata: BatchMetadata we want to associate with this DataAsset and all batches derived from it.
+
+        Returns:
+            The asset that is added to the datasource.
+        """
+        order_by_sorters: list[Sorter] = self.parse_order_by_sorters(order_by=order_by)
+        asset = PowerBIMeasure(
+            name=name,
+            order_by=order_by_sorters,
+            batch_metadata=batch_metadata or {},
+            groupby_columns=groupby_columns,
+            measure=measure,
+            # TODO: require custom serde for keys that are tuples
+            filters=filters,
+            fully_qualified_columns=fully_qualified_columns,
+            num_rows=num_rows,
+            pandas_convert_dtypes=pandas_convert_dtypes,
+            use_xmla=use_xmla,
+        )
+        return self._add_asset(asset)
+
+    def add_powerbi_table_asset(  # noqa: PLR0913
+        self,
+        name: str,
+        table: str,
+        order_by: Optional[SortersDefinition] = None,
+        batch_metadata: Optional[BatchMetadata] = None,
+        fully_qualified_columns: bool = False,
+        num_rows: Optional[int] = None,
+        multiindex_hierarchies: bool = False,
+        pandas_convert_dtypes: bool = True,
+    ) -> PowerBITable:
+        """Adds a PowerBITable asset to this datasource.
+
+        Args:
+            name: The name of this table asset.
+            table_name: The table where the data resides.
+            schema: The schema that holds the table.
+            order_by: A list of Sorters or Sorter strings.
+            batch_metadata: BatchMetadata we want to associate with this DataAsset and all batches derived from it.
+
+        Returns:
+            The asset that is added to the datasource.
+        """
+        order_by_sorters: list[Sorter] = self.parse_order_by_sorters(order_by=order_by)
+        asset = PowerBITable(
+            name=name,
+            order_by=order_by_sorters,
+            batch_metadata=batch_metadata or {},
+            table=table,
+            fully_qualified_columns=fully_qualified_columns,
+            num_rows=num_rows,
+            multiindex_hierarchies=multiindex_hierarchies,
+            pandas_convert_dtypes=pandas_convert_dtypes,
+        )
+        return self._add_asset(asset)
+
+    @staticmethod
+    def _running_on_fabric() -> bool:
+        if (
+            os.environ.get("AZURE_SERVICE", None)  # noqa: TID251 # needed for fabric
+            != "Microsoft.ProjectArcadia"
+        ):
+            return False
+        from pyspark.sql import SparkSession  # noqa: TID251 # needed for fabric
+
+        sc = SparkSession.builder.getOrCreate().sparkContext
+        return sc.getConf().get("spark.cluster.type") != "synapse"
```

Comment:
Same note here with these magic strings - if we put them as constants, we can add some context to them

Reply:
I've added a constant for `"Microsoft.ProjectArcadia"`, but I'm hesitant to do it here.

@eisber I copied the line on `394` from a snippet you provided, should this be `== "synapse" instead of `!= "synapse"`?

Reply:
I copied that code from our own code base. there's Synapse (PaaS) and Fabric (SaaS) and within Fabric there are also things called Synapse. code can't keep up with marketing.